packages/core/src/sync/index.ts

```diff
@@ -1167,6 +1167,10 @@ export async function* getLocalEventGenerator(params: {
         cursor = queryCursor;
         yield { events, checkpoint: cursor };
       } catch (error) {
+        if (params.common.shutdown.isKilled) {
+          throw new ShutdownError();
+        }
```

Comment:
In what scenario would this error not already be a ShutdownError ?

Reply:
If the query has a "statement timeout" error